### PR TITLE
docs(python): added all missing examples for temporal expressions

### PR DESCRIPTION
**Pull request by romanovacca**
Added examples for:
* strftime
* iso_year
* cast_time_zone
* tz_localize

Note: I did not add examples for some functions that are going to be unified together. See [#6445 ](https://github.com/pola-rs/polars/issues/6445)  for more info on this.

### CODE REVIEW

File: py-polars/polars/internals/expr/datetime.py [hunk 1]
```diff
@@ -383,6 +412,35 @@ def iso_year(self) -> pli.Expr:
         -------
         ISO Year as Int32
 
+        Examples
+        --------
+        >>> from datetime import timedelta, datetime
+        >>> start = datetime(2001, 1, 1)
+        >>> stop = datetime(2006, 1, 1)
+        >>> df = pl.DataFrame({"date": pl.date_range(start, stop, timedelta(days=180))})
```
Comment (stinodego) [inline py-polars/polars/internals/expr/datetime.py:420]: You seem to be a big fan of `date_range` 😄 I think the examples are fine, but would probably be clearer if you just hardcoded three datetimes in a dataframe. You already specified a start and stop, you're 2/3rds of the way there 😄

Comment (romanovacca) [reply py-polars/polars/internals/expr/datetime.py:420]: Yeah I could do that, but most of the other examples for expressions use date_range to make a couple examples, so I used this so it's in line with the other examples.

Comment (romanovacca-hema) [reply py-polars/polars/internals/expr/datetime.py:420]: But I agree I can trim down the amount of rows in this example, the amount now is a bit overkill

**Review by stinodego — APPROVED** [review]

**Comment by ritchie46** [comment]
Thank you @romanovacca and @stinodego for the review.

**Pull request merged**